queue/ArrayQueue.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Arrays;
+/**
+ * An array implementation of a queue
+ * 
+ * @author Harry Willis
+ */
+
+public class ArrayQueue<T> implements Queue<T>{
+    
+    private T[] queue;
+
+    private int front;
+    
+    // the next available spot in the queue
+    private int back;
+
+    private int size;
+
+    
+    /**
+     * Build default ArrayQueue with size 5
+     */
+    public ArrayQueue(){
+        this(5);
+    }
+
+    /**
+     * Build ArrayQueue with specified size
+     */
+    @SuppressWarnings("unchecked")
+    public ArrayQueue(int size){
```

Comment:
This is a stylistic thing but generally, it's nice to leave a space before the `{`

I like that you've been consistent with your spacing so if you prefer it this way then leave. Key is to be consistent.

Reply:
It's just habit. I like with the space more.